io/zenoh-transport/src/unicast/manager.rs

```diff
@@ -243,7 +243,7 @@ impl TransportManagerBuilderUnicast {
             protocols: Arc::new(AsyncMutex::new(HashMap::new())),
             transports: Arc::new(AsyncMutex::new(HashMap::new())),
             #[cfg(feature = "transport_multilink")]
-            multilink: Arc::new(MultiLink::make(prng)?),
+            multilink: Arc::new(MultiLink::make(prng, config.max_links > 1)?),
```

Comment:
I'd rather suggest not to create Multilink at all if multilink is not used: `Option<Multilink>` ....

Reply:
I agree with @yellowhatter 's suggestion. It's easier to have `Option<Multilink>` and to create `Some(multilink)` only when `config.max_links > 1`.

Reply:
Okay. I will revert the previous commits and change to this approach.